### PR TITLE
fix false positive error in spiller

### LINKED ISSUE #7560: Try to spill blocks containing only constant columns, it is meaningless to spill blocks containing only constant columns
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
["query id"=q1.sql] [errors="{\"Error 1105: other error for mpp stream: From MPP\\u003cquery:\\u003cquery_ts:1685215024530357702, local_query_id:103001, server_id:2277311, start_ts:441769007359983622\\u003e,task_id:63\\u003e: Code: 49, e.displayText() = DB::Exception: Check const_column_indexes.size() \\u003c input_schema.columns() failed: Try to spill blocks containing only constant columns, it is meaningless to spill blocks containing only constant columns, e.what() = DB::Exception,\":true,\"Error 1105: other error for mpp stream: From MPP\\u003cquery:\\u003cquery_ts:1685215036759783198, local_query_id:103133, server_id:2277311, start_ts:441769010558140419\\u003e,task_id:40\\u003e: Code: 0, e.displayText() = DB::Exception: Code: 0, e.displayText() = DB::Exception: Receiver state: ERROR, error message: Exchange receiver meet error : From MPP\\u003cquery:\\u003cquery_ts:1685215036759783198, local_query_id:103133, server_id:2277311, start_ts:441769010558140419\\u003e,task_id:62\\u003e: Code: 49, e.displayText() = DB::Exception: Check const_column_indexes.size() \\u003c input_schema.columns() failed: Try to spill blocks containing only constant columns, it is meaningless to spill blocks containing only constant columns, e.what() = DB::Exception,, e.what() = DB::Exception, e.what() = DB::Exception,\":true,\"Error 1105: other error for mpp stream: From MPP\\u003cquery:\\u003cquery_ts:1685215048037282265, local_query_id:103265, server_id:2277311, start_ts:441769013533474817\\u003e,task_id:63\\u003e: Code: 49, e.displayText() = DB::Exception: Check const_column_indexes.size() \\u003c input_schema.columns() failed: Try to spill blocks containing only constant columns, it is meaningless to spill blocks containing only constant columns, e.what() = DB::Exception,\":true}"]
```
### 4. What is your TiFlash version? (Required)
master

### PR DESCRIPTION
**Pull request by windtalker**
### What problem does this PR solve?

Issue Number: close #7560

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by yibin87 — APPROVED** [review]
LGTM

**Review by Lloyd-Pottiger — APPROVED** [review]

**Review by SeaRise — APPROVED** [review]

File: dbms/src/Core/Spiller.cpp [hunk 1]
```diff
@@ -78,6 +78,16 @@ void SpilledFiles::makeAllSpilledFilesImmutable()
     mutable_spilled_files.clear();
 }
 
+bool Spiller::supportSpill(const Block & header)
+{
+    for (size_t i = 0; i < header.columns(); ++i)
```
Comment (yibin87) [inline dbms/src/Core/Spiller.cpp:83]: change to range-for style?

Comment (windtalker) [reply dbms/src/Core/Spiller.cpp]: Done

**Pull request merged**